### PR TITLE
Make RestCompatTestTransformTask configuration cache compatible

### PR DESCRIPTION
**Pull request by breskeby**

### CODE REVIEW

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/rest/compat/compat/RestCompatTestTransformTask.java [hunk 1]
```diff
@@ -90,30 +91,28 @@ public class RestCompatTestTransformTask extends DefaultTask {
...
         this.sourceDirectory = objectFactory.directoryProperty();
         this.outputDirectory = objectFactory.directoryProperty();
-        this.testPatternSet = patternSetFactory.create();
+        this.testPatternSet = getPatternSetFactory().create();
         this.testPatternSet.include("/*" + "*/*.yml"); // concat these strings to keep build from thinking this is invalid javadoc
         // always inject compat headers
         headers.put("Content-Type", "application/vnd.elasticsearch+json;compatible-with=" + compatibleVersion);
         headers.put("Accept", "application/vnd.elasticsearch+json;compatible-with=" + compatibleVersion);
-        transformations.add(new InjectHeaders(headers, Set.of(RestCompatTestTransformTask::doesNotHaveCatOperation)));
+        getTransformations().add(new InjectHeaders(headers, Sets.newHashSet(RestCompatTestTransformTask::doesNotHaveCatOperation)));
```
Comment (breskeby) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/rest/compat/compat/RestCompatTestTransformTask.java:115]: refactor to use best practices and get rid of property and use `abstract  ListProperty getTransformations()` instead

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/rest/compat/compat/RestCompatTestTransformTask.java [hunk 2]
```diff
@@ -90,30 +91,28 @@ public class RestCompatTestTransformTask extends DefaultTask {
...
         this.compatibleVersion = Version.fromString(VersionProperties.getVersions().get("elasticsearch")).getMajor() - 1;
         this.sourceDirectory = objectFactory.directoryProperty();
         this.outputDirectory = objectFactory.directoryProperty();
-        this.testPatternSet = patternSetFactory.create();
+        this.testPatternSet = getPatternSetFactory().create();
         this.testPatternSet.include("/*" + "*/*.yml"); // concat these strings to keep build from thinking this is invalid javadoc
         // always inject compat headers
         headers.put("Content-Type", "application/vnd.elasticsearch+json;compatible-with=" + compatibleVersion);
         headers.put("Accept", "application/vnd.elasticsearch+json;compatible-with=" + compatibleVersion);
-        transformations.add(new InjectHeaders(headers, Set.of(RestCompatTestTransformTask::doesNotHaveCatOperation)));
```
Comment (breskeby) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/rest/compat/compat/RestCompatTestTransformTask.java:116]: Set.of creates an immutable collection that gradle configuration cache is choking on

**Review by mark-vieira — APPROVED** [review]
LGTM

**Comment by breskeby** [comment]
this work was related to https://github.com/elastic/elasticsearch/issues/57918

**Pull request merged**